app/controllers/users_controller.rb

```diff
@@ -1803,29 +1803,43 @@ def user_menu_messages
       raise Discourse::InvalidAccess.new("personal messages are disabled.")
     end
 
-    message_notifications = Notification.unread_type(
-      current_user,
-      Notification.types[:private_message],
-      USER_MENU_LIST_LIMIT
-    )
+    unread_notifications = Notification
+      .unread_types(
+        current_user,
+        [Notification.types[:private_message], Notification.types[:group_message_summary]],
+        USER_MENU_LIST_LIMIT
+      )
+      .prioritized(USER_MENU_LIST_LIMIT)
+      .to_a
 
-    if message_notifications.size < USER_MENU_LIST_LIMIT
-      exclude_topic_ids = message_notifications.map(&:topic_id).uniq
+    if unread_notifications.size < USER_MENU_LIST_LIMIT
+      exclude_topic_ids = unread_notifications.filter_map(&:topic_id).uniq
+      limit = USER_MENU_LIST_LIMIT - unread_notifications.size
       messages_list = TopicQuery.new(
         current_user,
-        per_page: USER_MENU_LIST_LIMIT - message_notifications.size
-      ).list_private_messages(current_user) do |query|
+        per_page: limit
+      ).list_private_messages_direct_and_groups(current_user) do |query|
         if exclude_topic_ids.present?
           query.where("topics.id NOT IN (?)", exclude_topic_ids)
         else
           query
         end
       end
+      read_notifications = Notification
+        .where(
+          read: true,
+          notification_type: Notification.types[:group_message_summary],
+          user_id: current_user.id
+        )
+        .visible
+        .includes(:topic)
+        .prioritized(limit)
```

Comment:
There is some overlap with the `unread_types` scope here. I'm wondering if we can introduce a generic scope instead and pass `read/unread` to the scope.

Reply:
I've done some refactoring on the scopes here. I think it's better now, but let me know what you think!